ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -172,22 +174,29 @@ private void clean(CompactionInfo ci, long minOpenTxnGLB, boolean metricsEnabled
       if (metricsEnabled) {
         perfLogger.perfLogBegin(CLASS_NAME, cleanerMetric);
       }
-      Table t = resolveTable(ci);
-      if (t == null) {
-        // The table was dropped before we got around to cleaning it.
-        LOG.info("Unable to find table " + ci.getFullTableName() + ", assuming it was dropped." +
+      String location = Optional.ofNullable(ci.properties).map(StringableMap::new)
```

Comment:
How this behaves with non-standard locations, and transformer generated locations.

Reply:
we only support this feature for acid tables with a non-customized location:
````
if (!TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType())) {
    if (tbl.getSd().getLocation() == null || tbl.getSd().getLocation().isEmpty()) {
        tblPath = wh.getDefaultTablePath(db, getPathWithSuffix(tbl.getTableName(), tbl), isExternal(tbl));
}

String getPathWithSuffix(String path, Table tbl) {
    return path + (tbl.isSetTxnId() ?
        VISIBILITY_PREFIX + String.format(DELTA_DIGITS, tbl.getTxnId()) : "");
}
````

Reply:
Maybe one more check - if the location is equals what we would have created -, so creating table with `show create table` would work as well.